alica_engine/include/engine/Types.h

```diff
@@ -63,7 +63,7 @@ using VariableGrp = std::vector<const Variable*>;
 
 using ParameterMap = std::unordered_map<std::string, Parameter*>;
 using AgentStatePair = std::pair<AgentId, const State*>;
-using TransitionConditionCallback = std::function<bool(const Blackboard*, const RunningPlan*, const IAlicaWorldModel*)>;
+using TransitionConditionCallback = std::function<bool(const Blackboard*, const RunningPlan*, const Blackboard*)>;
```

Comment:
can't use ref here? from one side inconsistent, but for c++ imho references is more preferable

Reply:
same in other places